lib/_stream_writable.js

```diff
@@ -302,7 +302,7 @@ function writeOrBuffer(stream, state, chunk, encoding, cb) {
 
   state.length += len;
 
-  var ret = state.length < state.highWaterMark;
+  var ret = state.length < state.highWaterMark && state.pendingcb < state.highWaterMark;
```

Comment:
I'm highly against it. Can you please add a unit test, or a script to reproduce the problem that this is fixing?